### PR TITLE
add additional information to message set acks

### PR DESCRIPTION
**Pull request by ripienaar**
I think there would be value in getting additional information about
acks specifically what set got it and what the sequence is:

 * A message can conceivably go to multiple set and callers will
   wait for all acks and want to ensure all got it - reference
   recent slack conversations, though I still think overlapping
   message sets are a bad idea

 * Someone will want to keep a audit trail of what messages in what
   sets belong to some transaction to later make it much easier to
   do GDPR erases over the data where you need the specific set IDs

 * For a kv type of use case the JetStreamMsgBySeqPre needs a set
   sequence so knowing this is valuable there too else I am not
   really sure how to get this easily

Kind of for discussion if this is a thing we want or not, I think we do - you had some concerns that this would be too much additional overhead.

### CODE REVIEW

File: server/msgset.go [hunk 1]
```diff
@@ -357,29 +357,29 @@ func (mset *MsgSet) processInboundJetStreamMsg(_ *subscription, _ *client, subje
 	doAck := !mset.config.NoAck
 	jsa := mset.jsa
 	stype := mset.config.Storage
+	sname := mset.config.Name
+	accName := c.acc.Name
```
Comment (ripienaar) [inline server/msgset.go]: whats cheaper 1 variable we wont use and later need to GC or getting an additional lock later when we need that? Anyway, seems removing a lock is a good thing so moved that up here

Comment (ripienaar) [reply server/msgset.go]: `c.acc.Name` surely cant change? So didnt need to be in the lock I think - and was used directly unlocked in the limits check - so I set it outside of the lock now after the c != nil check

**Comment by ripienaar** [comment]
Should I tweak this to have a option to enable it? Personally I prefer it on by default these acks are tiny and so does not seem worth adding to the config fatigue.

**Comment by derekcollison** [comment]
Need to think more about this, on my list.

**Pull request closed without merging**